front/src/applications/rollingStockEditor/components/RollingStockEditorForm.tsx

```diff
@@ -420,25 +385,72 @@ const RollingStockEditorForm = ({
     );
   };
 
-  // TODO: use InputSNCF and InputGroupSNCF
+  useEffect(() => {
+    setIsValid(true);
+    RollingStockschemaProperties.forEach((property) => {
+      if (
+        (property.title === Parameter[property.title as Parameter] &&
+          Number.isNaN(rollingStockValues[property.title])) ||
+        (rollingStockValues[property.title] as number) < (property.min as number) ||
+        (rollingStockValues[property.title] as number) > (property.max as number)
+      ) {
+        setIsValid(false);
+      }
+    });
+  }, [rollingStockValues]);
+
+  useEffect(() => {
+    if (rollingStockData) {
+      setCurrentRsEffortCurve(rollingStockData.effort_curves);
+      setIsCurrentEffortCurveDefault(false);
+    }
+  }, [rollingStockData]);
 
   return (
     <form
       className="d-flex flex-column form-control rollingstock-editor-form bg-white"
-      onSubmit={handleSubmit(submit)}
+      onSubmit={(e) => submit(e, rollingStockValues)}
     >
-      <div className="d-lg-flex justify-content-center mb-4">
-        <div className="col-lg-6 mr-4">{metadataForm(sideValue.left, register)}</div>
-        <div className="col-lg-6">{metadataForm(sideValue.right, register)}</div>
+      <div className="d-lg-flex justify-content-center mb-4 px-1">
+        <div className="col-lg-4 mr-4 rollingstock-editor-input-container">
+          {metadataForm(
+            RollingStockschemaProperties.filter((schemaProperty) => schemaProperty.side === 'left')
```

Comment:
You can remove the filter in metadataForm if you filter all the fields here: 
```suggestion
            RollingStockschemaProperties.filter((schemaProperty) => schemaProperty.side === 'left' && Metadata.includes(schemaProperty.title)
```